src/ai/backend/manager/models/alembic/versions/1d42c726d8a3_add_container_registry_tables.py

```diff
@@ -0,0 +1,241 @@
+"""Migrate container registry schema from `etcd` to `postgreSQL`
+
+Revision ID: 1d42c726d8a3
+Revises: 75ea2b136830
+Create Date: 2024-03-05 10:36:24.197922
+
+"""
+
+import asyncio
+import json
+import os
+import sys
+from concurrent.futures import ThreadPoolExecutor
+from itertools import groupby
+from typing import Any, Final, Mapping, cast
+
+import janus
+import sqlalchemy as sa
+import trafaret as t
+from alembic import op
+
+from ai.backend.common import validators as tx
+from ai.backend.common.etcd import AsyncEtcd, ConfigScopes
+from ai.backend.manager.config import load
+from ai.backend.manager.models.base import IDColumn, StrEnumType, convention
+from ai.backend.manager.models.container_registry import ContainerRegistryRow, ContainerRegistryType
+
+# revision identifiers, used by Alembic.
+revision = "1d42c726d8a3"
+down_revision = "75ea2b136830"
+branch_labels = None
+depends_on = None
+
+etcd_container_registry_iv = t.Dict({
+    t.Key(""): tx.URL,
+    t.Key("type", default="docker"): t.String,
+    t.Key("username", default=None): t.Null | t.String,
+    t.Key("password", default=None): t.Null | t.String(allow_blank=True),
+    t.Key("project", default=None): (
+        t.Null | t.List(t.String) | tx.StringList(empty_str_as_empty_list=True)
+    ),
+    tx.AliasedKey(["ssl_verify", "ssl-verify"], default=True): t.ToBool,
+}).allow_extra("*")
+
+ETCD_CONTAINER_REGISTRY_KEY: Final = "config/docker/registry"
+
+ETCD_CONTAINER_REGISTRIES_BACKUP_FILENAME: Final = "etcd_container_registries_backup.json"
+
+
+def get_async_etcd() -> AsyncEtcd:
+    local_config = load()
+    etcd_config = local_config.get("etcd", None)
+    assert etcd_config is not None, "etcd configuration is not found"
+    etcd_credentials = None
+    if local_config["etcd"]["user"]:
+        etcd_credentials = {
+            "user": local_config["etcd"]["user"],
+            "password": local_config["etcd"]["password"],
+        }
+    scope_prefix_map = {
+        ConfigScopes.GLOBAL: "",
+    }
+    return AsyncEtcd(
+        local_config["etcd"]["addr"],
+        local_config["etcd"]["namespace"],
+        scope_prefix_map,
+        credentials=etcd_credentials,
+    )
+
+
+def migrate_data_etcd_to_psql():
+    queue = janus.Queue()
```

Comment:
Let's add explicit type annotations here for future readers and editors.

Reply:
In your implementation, you don't interface this queue between non-async and async codes, but uses the queue to store the results from _non-async_ code in separate executor threads running in parallel.
As such, we can just use the stdlib's queue for simpler code!